httpmw/logging_mw.go

```diff
@@ -0,0 +1,61 @@
+package httpmw
+
+import (
+	"net/http"
+	"time"
+
+	"github.com/gorilla/mux"
+
+	"github.com/Azure/aks-middleware/ctxlogger"
+)
+
+func NewLogging() mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return &loggingMiddleware{
+			next: next,
+			now:  time.Now,
+		}
+	}
+}
+
+var _ http.Handler = &loggingMiddleware{}
+
+type loggingMiddleware struct {
+	next http.Handler
+	now  func() time.Time
+}
+
+type responseWriter struct {
+	http.ResponseWriter
+	statusCode int
+}
+
+func (rw *responseWriter) WriteHeader(code int) {
+	rw.statusCode = code
+	rw.ResponseWriter.WriteHeader(code)
+}
+
+func (rw *responseWriter) Write(b []byte) (int, error) {
+	if rw.statusCode == 0 {
+		rw.statusCode = http.StatusOK
+	}
+	return rw.ResponseWriter.Write(b)
+}
+
+func (l *loggingMiddleware) ServeHTTP(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	logger := ctxlogger.GetLogger(ctx)
+
+	customWriter := &responseWriter{ResponseWriter: w}
+
+	startTime := l.now()
+	l.next.ServeHTTP(customWriter, r)
+	endTime := l.now()
+
+	latency := endTime.Sub(startTime)
+
+	logger.With(
```

Comment:
Can you use an interface method like `logRequestStart` and `logRequestEnd` to specify that the call is started/finished respectively? Different services log requests differently, and using an interface method would allow users to plumb in whatever implementation they desire.

From the aks-rp side, we use the messages "***RequestStart" and "***RequestEnd" to designate them, and a lot of our alerts filter off of those messages. Using an interface would allow basically a drop in replacement, without the need to go and change any other alerts